Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-contests.tsx

```diff
@@ -1,70 +1,55 @@
 import * as React from 'react';
 import { createContext, useCallback, useContext, useEffect, useState } from 'react';
 import { AxiosResponse } from 'axios';
-import IHaveChildrenProps from '../../components/common/IHaveChildrenProps';
-import { getIndexContestsUrl, getProblemResourceUrl, startContestParticipationUrl } from '../../utils/urls';
-import { useHttp } from '../use-http';
-import { useLoading } from '../use-loading';
+import { IHaveChildrenProps } from '../components/common/Props';
+import { getIndexContestsUrl, getProblemResourceUrl } from '../utils/urls';
+import { useHttp } from './use-http';
+import { useLoading } from './use-loading';
 import {
-    IContestType,
     IProblemType,
     IIndexContestsType,
     IGetContestsForIndexResponseType,
-    IStartParticipationResponseType,
     ISubmissionTypeType,
-} from './types';
-
-interface IContestsContext {
-    currentContest: IContestType | null,
-    isContestParticipationOfficial: boolean,
-    currentProblem: IProblemType | null,
-    selectedSubmissionTypeId: number | null,
-    setProblem: (problem: IProblemType) => void,
-    setSubmissionType: (id: number) => void,
-    activeContests: IIndexContestsType[]
-    pastContests: IIndexContestsType[]
+} from '../common/types';
+
+interface IContestsContext2 {
+    currentProblem: IProblemType | null;
+    selectedSubmissionTypeId: number | null;
+    setProblem: (problem: IProblemType) => void;
+    setSubmissionType: (id: number) => void;
+    activeContests: IIndexContestsType[];
+    pastContests: IIndexContestsType[];
     getForHome: () => Promise<void>;
-    startContestParticipation: (id: number, isOfficial: boolean) => Promise<void>;
     getProblemResourceFile: (resourceId: number) => Promise<void>;
     getProblemResourceResponse: AxiosResponse;
```

Comment:
This is not wrapped in an abstraction?

Reply:
Good point.
This will be fixed here: https://github.com/SoftUni-Internal/exam-systems-issues/issues/142